File: src/test_bridge_gateway.py
#!/usr/bin/env python3
"""Test Konoha Bridge Router database schemas and model prefix resolution."""
import os
import sqlite3
import unittest
from unittest.mock import patch, MagicMock

DB_PATH = os.path.expanduser("~/.konoha/skills.db")

class TestBridgeGateway(unittest.TestCase):
    def setUp(self):
        self.assertTrue(os.path.exists(DB_PATH), "Database missing. Run migration first.")

    def test_bridges_schema(self):
        """Verify the bridges table has the correct structure."""
        conn = sqlite3.connect(DB_PATH)
        cursor = conn.cursor()
        cursor.execute("PRAGMA table_info(bridges);")
        columns = {row[1]: row[2] for row in cursor.fetchall()}
        conn.close()

        expected_columns = ["name", "port", "provider", "enabled", "target_url", "api_key"]
        for col in expected_columns:
            self.assertIn(col, columns, f"Column '{col}' is missing from bridges table.")

    def test_bridge_creation_and_retrieval(self):
        """Test database bridge insertion and retrieval operations."""
        conn = sqlite3.connect(DB_PATH)
        cursor = conn.cursor()
        
        # Insert a dummy bridge
        try:
            cursor.execute(
                "INSERT OR REPLACE INTO bridges (name, port, provider, enabled, target_url, api_key) "
                "VALUES ('test_bridge', 12345, 'openai', 1, 'http://localhost:12345', 'dummy_key');"
            )
            conn.commit()

            cursor.execute("SELECT port, provider FROM bridges WHERE name='test_bridge';")
            row = cursor.fetchone()
            self.assertIsNotNone(row)
            self.assertEqual(row[0], 12345)
            self.assertEqual(row[1], "openai")

            # Clean up test bridge
            cursor.execute("DELETE FROM bridges WHERE name='test_bridge';")
            conn.commit()
        except sqlite3.Error as e:
            self.fail(f"Bridge insert/delete test failed: {e}")
        finally:
            conn.close()

if __name__ == "__main__":
    unittest.main()

File: tests/test_bridge_gateway.js
#!/usr/bin/env node
'use strict';

/**
 * tests/test_bridge_gateway.js — Test Konoha Bridge Router database schemas and operations.
 */

const assert = require('assert');
const fs = require('fs');
const os = require('os');
const path = require('path');
const db = require('../src/db');

async function run() {
  console.log('Running test_bridge_gateway tests...');
  const dbPath = db.DB_PATH;
  assert.ok(fs.existsSync(dbPath), `Database missing: ${dbPath}. Run migration first.`);

  const conn = db.getDb(dbPath);
  try {
    // 1. Bridges schema
    const cols = conn.prepare('PRAGMA table_info(bridges);').all();
    const colMap = {};
    for (const c of cols) {
      colMap[c.name] = c.type;
    }
    const expectedColumns = ['name', 'port', 'provider', 'enabled', 'target_url', 'api_key'];
    for (const col of expectedColumns) {
      assert.ok(col in colMap, `Column '${col}' is missing from bridges table.`);
    }
    console.log('✓ Bridges schema verified');

    // 2. Bridge creation and retrieval
    conn.prepare(`
      INSERT OR REPLACE INTO bridges (name, port, provider, enabled, target_url, api_key)
      VALUES ('test_bridge', 12345, 'openai', 1, 'http://localhost:12345', 'dummy_key')
    `).run();

    const row = conn.prepare("SELECT port, provider FROM bridges WHERE name='test_bridge'").get();
    assert.ok(row, 'Bridge row must exist');
    assert.strictEqual(row.port, 12345);
    assert.strictEqual(row.provider, 'openai');

    // Clean up
    conn.prepare("DELETE FROM bridges WHERE name='test_bridge'").run();
    const rowAfter = conn.prepare("SELECT port, provider FROM bridges WHERE name='test_bridge'").get();
    assert.strictEqual(rowAfter, undefined);
    console.log('✓ Bridge creation and retrieval verified');

    console.log('\nAll test_bridge_gateway tests passed!');
  } finally {
    conn.close();
  }
}

if (require.main === module) {
  run().catch(err => {
    console.error(err);
    process.exit(1);
  });
}

module.exports = { run };
